### ISSUE TITLE
Can Build a Wall covering Wall-Mounts

### ISSUE THREAD

**Issue by cosmiccoincidence** [RE-SS3D/SS3D#1044]
## Summary

You can build a wall blocking a wall mount and the wall mount will still be there when you remove the wall again.

## Steps to Reproduce

- Place a wall mount.
- Place a wall on the tile in front of the wall mount.
- Delete that wall, and notice the wall mount is still in place and was being hidden by the wall model.

## Expected Behavior

If a wall mount gets blocked by a new wall, delete the wall mount.

## Context

https://user-images.githubusercontent.com/41941008/221445891-2f6cb596-1541-4a49-b030-c61678557417.mp4

**Comment by stilnat** [comment]
wouldn't the expected behavior be that it should not allow the wall to be built ? We could use the Tilemap system for the construction system in game, and it feels weird to me that building a wall would just destroy the wall mount. But if that's really the expected behavior I'm fine with it, just wondering.

**Comment by cosmiccoincidence** [comment]
> wouldn't the expected behavior be that it should not allow the wall to be built ? 

In the in-game construction we will prevent a wall from being built if there is a wallmount obstructing it, but for the map editor it would be nice to have this overwrite feature so you dont need to delete wall fixtures prior to making a wall there. But, this instance may not happen that often so I'm fine with either method.
